dxr/utils.py

```diff
@@ -285,3 +285,7 @@ def is_in(needle, haystack):
 def without_ending(ending, string):
     """If ``string`` ends with ``ending``, strip it off."""
     return string[:-len(ending)] if string.endswith(ending) else string
+
+def splitlines_keepends(string):
```

Comment:
Rather than naming this for what it mechanically does, let's name it for its intended use. That way, if we decide to change the mechanics later, it doesn't have to get renamed. Something like `split_content_lines` might be appropriate.